### PR TITLE
Avoid false positives on egl extension matching

### PR DESCRIPTION
**Pull request by agx**
Due to the strstr prefix match EGL_EXT_foo would be incorrectly matched
if EGL_EXT_foobar would be available but not foo.

This doesn't matter for the currently checked extensions but will matter
for EGL_EXT_image_dma_buf_import_modifiers vs
EGL_EXT_image_dma_buf_import

Code borrowed from weston

### CODE REVIEW

File: render/egl.c [hunk 1]
```diff
@@ -83,6 +83,26 @@ static bool egl_get_config(EGLDisplay disp, EGLint *attribs, EGLConfig *out,
...
 }
 
+static bool
+check_egl_ext(const char *egl_exts, const char *ext)
+{
+	size_t extlen = strlen(ext);
+	const char *end = egl_exts + strlen(egl_exts);
+
+	while (egl_exts < end) {
+		size_t n = 0;
```
Comment (emersion) [inline render/egl.c]: No need to delcare `n` here, declare it when first defined.

**Comment by agx** [comment]
Missed that one. Updated.

File: render/egl.c [hunk 2]
```diff
@@ -83,6 +83,25 @@ static bool egl_get_config(EGLDisplay disp, EGLint *attribs, EGLConfig *out,
 	return false;
 }
 
+static bool
+check_egl_ext(const char *egl_exts, const char *ext)
+{
```
Comment (emersion) [inline render/egl.c]: Style: `static bool check_egl_ext(…) {` on the same line

File: render/egl.c [hunk 3]
```diff
@@ -83,6 +83,25 @@ static bool egl_get_config(EGLDisplay disp, EGLint *attribs, EGLConfig *out,
...
+	size_t extlen = strlen(ext);
+	const char *end = egl_exts + strlen(egl_exts);
+
+	while (egl_exts < end) {
+		if (*egl_exts == ' ') {
+			egl_exts++;
+			continue;
+		}
+		size_t n = strcspn(egl_exts, " ");
+		if (n == extlen && strncmp(ext, egl_exts, n) == 0)
```
Comment (emersion) [inline render/egl.c]: Style: no conditions without brackets

**Comment by agx** [comment]
On Wed, Feb 28, 2018 at 10:11:45AM -0800, emersion wrote:
> emersion commented on this pull request.
> 
> 
> 
> > @@ -83,6 +83,25 @@ static bool egl_get_config(EGLDisplay disp, EGLint *attribs, EGLConfig *out,
>  	return false;
>  }
>  
> +static bool
> +check_egl_ext(const char *egl_exts, const char *ext)
> +{
> +	size_t extlen = strlen(ext);
> +	const char *end = egl_exts + strlen(egl_exts);
> +
> +	while (egl_exts < end) {
> +		if (*egl_exts == ' ') {
> +			egl_exts++;
> +			continue;
> +		}
> +		size_t n = strcspn(egl_exts, " ");
> +		if (n == extlen && strncmp(ext, egl_exts, n) == 0)
> 
> Style: no conditions without brackets

I'm having a hard time adjusting to this coding style, fixed now.

File: render/egl.c [hunk 4]
```diff
@@ -83,6 +83,25 @@ static bool egl_get_config(EGLDisplay disp, EGLint *attribs, EGLConfig *out,
 	return false;
 }
 
+static bool
+check_egl_ext(const char *egl_exts, const char *ext) {
```
Comment (emersion) [inline render/egl.c]: Style: `static bool` on the same line! :P

**Comment by agx** [comment]
On Wed, Feb 28, 2018 at 06:48:59PM +0000, emersion wrote:
> emersion commented on this pull request.
> 
> 
> 
> > @@ -83,6 +83,25 @@ static bool egl_get_config(EGLDisplay disp, EGLint *attribs, EGLConfig *out,
>  	return false;
>  }
>  
> +static bool
> +check_egl_ext(const char *egl_exts, const char *ext) {
> 
> Style: `static bool` on the same line! :P

Sorry about that. It's just so different from my habit, updated.

**Comment by emersion** [comment]
Feel free to join our IRC channel if you want to discuss about the project. Thanks!

**Pull request merged**